client/src/components/MapView.vue

```diff
@@ -95,9 +106,7 @@ export default defineComponent({
     updateMapOnDataLayerChange(newDataLayer: DataLayer | null, oldDataLayer: DataLayer | null): void {
       if (this.map == null) {
         const source = newDataLayer?.source;
-        const dataLoaded =
-          (source?.type == DataSourceType.GeoJson && source?.data != null) ||
-          (source?.type == DataSourceType.Vector && source?.tiles != null);
+        const dataLoaded = source?.type == DataSourceType.Vector && source?.tiles != null;
```

Comment:
nit: I think we can remove the source.type check since all data sources will now be vector

Reply:
Removed in the condition. We have to keep it overall because the types for mapbox require type

Reply:
Just kidding once I removed it complained that only Vector types have tiles so the condition had to be there